src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ListViewModel.cs

```diff
@@ -18,20 +19,45 @@ public partial class ListViewModel : ObservableObject
     [ObservableProperty]
     public partial ObservableGroupedCollection<string, ListItemViewModel> Items { get; set; } = [];
 
+    private readonly ExtensionObject<IListPage> _model;
+
+    private readonly TaskScheduler _taskScheduler;
+
     public ListViewModel(IListPage model)
     {
-        // TEMPORARY: just plop all the items into a single group
-        // see 9806fe5d8 for the last commit that had this with sections
-        ObservableGroup<string, ListItemViewModel> group = new(string.Empty);
-
-        foreach (var item in model.GetItems())
-        {
-            ListItemViewModel viewModel = new(item);
-            _ = viewModel.InitializePropertiesAsync();
-            group.Add(viewModel);
-        }
-
-        Items.AddGroup(group);
+        _model = new(model);
+        _taskScheduler = TaskScheduler.FromCurrentSynchronizationContext();
```

Comment:
We may want to pass this in via the View in the constructor, that way if we ever change the initialization, we'll know that's a hard requirement to construct the object.